Core/Versioning/KspVersionCriteria.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace CKAN.Versioning
+{
+    public class KspVersionCriteria
+    {
+        private List<KspVersion> versions = new List<KspVersion> ();
+
+        public KspVersionCriteria (KspVersion v)
+        {
+            this.versions.Add (v);
+        }
+
+        public KspVersionCriteria(KspVersion v, List<KspVersion> compatibleVersions)
+        {
+            if(v != null) { 
+                this.versions.Add(v);
+            }
+            this.versions.AddRange(compatibleVersions);
+        }
+
+        public List<KspVersion> Versions {
```

Comment:
Actually, I'd rather have this class implement `IEnumerable<KspVersion>` and get rid of the `Versions` property. It'd simplify a lot of code using this.